assets/src/hooks/useFetchTimepoints.ts

```diff
@@ -1,26 +1,44 @@
-import { useEffect } from "react"
+import { useEffect, useState } from "react"
 import { fetchTimepointsForRoute } from "../api"
 import { RouteId, Timepoint, TimepointsByRouteId } from "../skate"
-import {
-  Dispatch,
-  setLoadingTimepointsForRoute,
-  setTimepointsForRoute,
-} from "../state"
 
-export const useFetchTimepoints = (
-  selectedRouteIds: RouteId[],
-  timepointsByRouteId: TimepointsByRouteId,
-  dispatch: Dispatch
-): void => {
+const useFetchTimepoints = (
```

Comment:
Since this is now a hook, what about `useTimepoints(RouteId): Timepoint[] | null` instead? The map seems like overkill.

Reply:
React internals depend on hooks always being called at the top level. You can't put them in a loop. [explanation](https://reactjs.org/docs/hooks-rules.html)

You could call the hook with a single argument if the hook was called directly from the RouteLadder component instead of from the App component, which is similar to how we used to do it, but I like having the data live top level.

Reply:
It wouldn't be in a loop: each component which needed the timepoint data would call `const timepoints = useTimepoints(routeId)` with the routeId it receives as an argument.

Reply:
I want to keep it at the top level so that e.g. if the Ladder and the VehicleProperties components both need the timepoints, they can use the same data rather than each having to do their own fetch.

Reply:
Isn't avoiding an extra fetch the responsibility of the hook? Both of the components would use `const timepoints = useTimepoints(routeId)` and the hook gives them the data.

Reply:
Hooks don't really have their own state. They store their state in their component. So if two separate components use the same hook, those two instances of the hook wouldn't share any state.

I suppose the hook could abstract away a reference to some shared mutable state, but that defeats the purpose of using React / FRP.